server/team/team.contoller.js

```diff
@@ -125,16 +123,22 @@ function list(req, res) {
 function single(req, res) {
   return Team.findOne({ _id: req.params.teamId }).then((team) => {
     if (!team) {
-      return res.redirect('/team');
+      return res.status(200).json(['']);
     }
     return res.status(200).json(team);
   });
 }
 
 function deleteTeam(req, res) {
-  return Team.remove({ _id: req.params.teamId }).then(team =>
-    res.status(200).json(`deleted team ${team.name}`)
-  );
+  return Team.remove({ _id: req.params.teamId }).then(() => {
+    User.update({ teamId: req.params.teamId }, { $set: { teamId: null } })
+    .then(() => {
+      Webhook.remove({ belongsTo: req.params.teamId })
+      .then(() => {
+        res.status(200).json('deleted team');
```

Comment:
We need to redirect here. The user is left on a page with the ability to update a team that doesn't exist. That will only cause bad things to happen if someone of the right mind figures it out.

The home page would be a good place to redirect to.